### DISCUSSION TITLE
aliases not working in preact config file fopr preact typescript

### DISCUSSION THREAD

**Discussion by ljmerza** [preactjs/preact-cli#1697] [Q&A]
in the config file i have:

```
export default (config, env, helpers, options) => {
  config.resolve.alias = {
    'components': 'src/components',
    ...config.resolve.alias
  }
}
```

then in a component i have `import { Header } from 'components'`  but i get the error:

```
ERROR in ./components/app.tsx
Module not found: Error: Can't resolve 'components' in '/home/user/projects/repo/src/components'
```

ive also tried

```
export default (config, env, helpers) => {
    config.resolve.alias.components= 'src/components';
}
```

and 

```
export default (config, env, helpers) => {
    config.resolve.modules.push('src/components');
}
```

**Comment by rschristian** [comment]
Please check out our config recipes wiki, it has solutions to this issue and more.

https://github.com/preactjs/preact-cli/wiki/Config-Recipes#absolute-imports

**Reply by ljmerza** [reply]
Hey, did you go through my post? I've shown the two examples right there and they didn't work.

**Reply by rschristian** [reply]
No, you did not.

`'src/components'` is not a valid file path. The alias does nothing as it's invalid.

You can use `env.src` as we've written in the config wiki, or build an absolute file path.

**Reply by rschristian** [reply]
Sorry, I imagined you hadn't as your provided examples were different from our own, and, rather importantly, our config actually works.

Just copy/pasting it would've fixed your issue.

As for file paths, that's just a Webpack thing, nothing specific to us.

**Reply by ljmerza** [reply]
building an absolute path was the answer i was looking for instead of mearly linking what i already looked at. the example should be

```
config.resolve.alias.components= '${env.src}/components`;

``
